src/Symfony/Component/Config/Resource/FileResource.php

```diff
@@ -29,10 +29,16 @@ class FileResource implements SelfCheckingResourceInterface, \Serializable
      * Constructor.
      *
      * @param string $resource The file path to the resource
+     *
+     * @throws \InvalidArgumentException
      */
     public function __construct($resource)
     {
         $this->resource = realpath($resource);
+
+        if (false === $this->resource) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not exist.', $resource));
```

Comment:
Well, for eval'd classes, the code was already weird before as it was creating a FileResource for something which is not a file. so I would consider it a bug in the ObjectRouteLoader